Source/WriteableBitmapEx/BitmapContext.cs

```diff
@@ -105,38 +108,59 @@ public BitmapContext(WriteableBitmap writeableBitmap, ReadWriteMode mode)
         {
             _writeableBitmap = writeableBitmap;
             _mode = mode;
-
+#if !WPF
             _pixelWidth = _writeableBitmap.PixelWidth;
             _pixelHeight = _writeableBitmap.PixelHeight;
+#endif
 #if WPF
          //// Check if it's the Pbgra32 pixel format
          //if (writeableBitmap.Format != PixelFormats.Pbgra32)
          //{
          //   throw new ArgumentException("The input WriteableBitmap needs to have the Pbgra32 pixel format. Use the BitmapFactory.ConvertToPbgra32Format method to automatically convert any input BitmapSource to the right format accepted by this class.", "writeableBitmap");
          //}
 
-            double width = _writeableBitmap.BackBufferStride / WriteableBitmapExtensions.SizeOfArgb;
-            _length = (int)(width * _pixelHeight);
+            BitmapContextBitmapProperties bitmapProperties;
 
             lock (UpdateCountByBmp)

```

Comment:
Does this really need to be locked? It's a Concurrent Dictionary anyway, so it handles the thread-safety already. Right?